test/units/module_utils/facts/test_collectors.py

```diff
@@ -218,6 +219,24 @@ class TestNetworkCollector(BaseFactsTest):
     collector_class = NetworkCollector
 
 
+class TestOSReleaseFacts(BaseFactsTest):
+    __test__ = True
+    gather_subset = ['!all', 'os_release']
+    valid_subsets = ['os_release']
+    fact_namespace = 'ansible_os_release'
+    collector_class = OSReleaseFactCollector
+
+    @patch('ansible.module_utils.facts.system.os_release.get_file_content', return_value='# test\nID=clown_linux\nVERSION="Clown Linux"')
```

Comment:
I'd like to see more tests of the parsing, ideally with some examples of real os-release files. test/units/module_utils/test_distribtution.py[1] has some.

[1] test_distribution.py file is in the wrong place but nonetheless